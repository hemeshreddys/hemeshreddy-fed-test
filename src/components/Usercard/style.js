import Styled from 'styled-components'

export const Card = Styled.div `
    display: flex;
    flex-direction: row;
    align-items: center;
    width: 100%;
    margin-bottom:5px;
    padding: 10px 20px;
    background-color:#f44336;
    border-radius: 5px;
    color: #ffffff;    
    @media (min-width: 1024px) {
        width: 30%;
    }
    img{
        border-radius: 75px;
        max-width: 25%;
    }
    .users{
        margin-left:20px;
        h2, p { 
            margin: 0;
        }
    }
`
export const Container = Styled.div `
    display: flex;
    width:100%;
    flex-direction: column;
    justify-content: center;
    align-items: center;
`

export const Btn = Styled.a `
    color: #ffffff;
    background-color: #f44336;
    padding: 10px 20px;
    text-decoration: none;
    border-radius: 5px;
`