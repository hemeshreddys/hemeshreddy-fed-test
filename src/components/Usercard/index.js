import React, { useEffect, useState } from "react";
import axios from "axios";
import * as style from "./style.js";
import CardItem from '../CardItem'

const Usercard = () => {
  const [users, setUsers] = useState([]);
  const [page, setPage] = useState(1);
  const [data, setData] = useState([]);

  const loadMoreData = () => {
    let url = `https://randomuser.me/api/?results=4&page=${page}`;
    axios.get(url).then((res) => {
      setData([...data, ...res.data.results]);
      setPage(page + 1);
    });
  };

  useEffect(() => {
    axios("https://randomuser.me/api/?results=4")
      .then((response) =>
        response.data.results.map((user) => ({
          name: `${user.name.first} ${user.name.last}`,
          email: `${user.email}`,
          image: `${user.picture.large}`,
          location: `${user.location.state}`,
        }))
      )
      .then((data) => {
        setUsers(data);
      });
  }, []);

  return (
      <style.Container>
        {users.map((user,index) => (
            <CardItem key={index} imgurl={user.image} name={user.name} email={user.email} location={user.location}/> 
        ))}
        {data.map((article, index) => (
            <CardItem key={index} imgurl={article.picture.large} name={article.name.first + article.name.last} email={article.email} location={article.location.state}/> 
        ))}
        <style.Btn className="btn-primary" href="#" onClick={loadMoreData}>
          Load more
        </style.Btn>
      </style.Container>
  );
};

export default Usercard;
