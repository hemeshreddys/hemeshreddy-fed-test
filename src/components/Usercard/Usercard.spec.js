import React from 'react';
import Usercard from './index.js';

describe('Usercard', () => {
	it('renders the Usercard', () => {
		const actual = mount(<Usercard />);
		expect(actual.get(0)).toMatchSnapshot();
		
	});
});
