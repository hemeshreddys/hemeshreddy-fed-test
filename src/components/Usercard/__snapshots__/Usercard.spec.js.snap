// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Usercard renders the Usercard 1`] = `<Usercard />`;
